draft-irtf-cfrg-vdaf.md

```diff
@@ -2255,65 +2264,69 @@ i.e., when `Flp.JOINT_RAND_LEN == 0`:
-        )
-
-    # Each Aggregator's input share contains its measurement share
-    # and proof share.
-    input_shares = []
-    input_shares.append((
-        leader_meas_share,
-        leader_proof_share,
-        None,
-    ))
-    for j in range(Prio3.SHARES-1):
-        input_shares.append((
-            k_helper_meas_shares[j],
-            k_helper_proof_shares[j],
-            None,
-        ))
-    return (None, input_shares)
+  k_helper_seeds, seeds = front((Prio3.SHARES-1) * 2, seeds)
+  k_helper_meas_shares = [
+      k_helper_seeds[i]
+      for i in range(0, (Prio3.SHARES-1) * 2, 2)
+  ]
+  k_helper_proofs_shares = [
+      k_helper_seeds[i]
+      for i in range(1, (Prio3.SHARES-1) * 2, 2)
+  ]
+  (k_prove,), seeds = front(1, seeds)
+
+  # Shard the encoded measurement into shares.
+  leader_meas_share = meas
+  for j in range(Prio3.SHARES-1):
+      leader_meas_share = vec_sub(
+          leader_meas_share,
+          Prio3.helper_meas_share(j+1, k_helper_meas_shares[j]),
+      )
+
+  # Generate the proofs and shard it into proof(s) shares.
+  prove_rands = Prio3.prove_rands(k_prove)
+  leader_proofs_share = []
+  for _ in range(Prio3.PROOFS):
+      prove_rand, prove_rands = front(
+          Prio3.Flp.PROVE_RAND_LEN, prove_rands)
+      leader_proofs_share += Prio3.Flp.prove(meas, prove_rand, [])
+  for j in range(Prio3.SHARES-1):
+      leader_proofs_share = vec_sub(
+          leader_proofs_share,
+          Prio3.helper_proofs_share(j+1, k_helper_proofs_shares[j]),
+      )
+
+  # Each Aggregator's input share contains its measurement share
+  # and proofs share.
+  input_shares = []
+  input_shares.append((
+      leader_meas_share,
+      leader_proofs_share,
+      None,
+  ))
+  for j in range(Prio3.SHARES-1):
+      input_shares.append((
+          k_helper_meas_shares[j],
+          k_helper_proofs_shares[j],
+          None,
+      ))
+  return (None, input_shares)
 ~~~
 {: #prio3-shard-without-joint-rand title="Sharding an encoded measurement without joint randomness."}
 
 The steps in this method are as follows:
 
 1. Shard the encoded measurement into shares
-1. Generate the proof and shard the proof into shares
-1. Encode each measurement and proof share into an input share
+1. Generate the proofs and shard the proofs into shares
+1. Encode each measurement and proofs share into an input share
 
-Notice that only one pair of measurement and proof shares (called the "leader"
+Notice that only one pair of measurement and proofs shares (called the "leader"
 shares above) are vectors of field elements. The other shares (called the
 "helper" shares) are represented instead by XOF seeds, which are expanded into
 vectors of field elements.
 
-The methods on `Prio3` for deriving the prover randomness, measurement shares,
-and proof shares and the methods for encoding the input shares are defined in
+The methods on `Prio3` for deriving the prover randomnesses, measurement shares,
```

Comment:
"randomnesses" might be the right term if "randomness" would be ambiguous in context. Here I don't think it is. (Similar comment below.)

```suggestion
The methods on `Prio3` for deriving the prover randomness, measurement shares,
```